modules/pico_mdns.c

```diff
@@ -2356,7 +2404,12 @@ pico_mdns_additionals_add_nsec( pico_mdns_rtree *artree,
 
     /* If there is none present generate one for given name */
     if ((nsec = pico_mdns_gen_nsec_record(name)))
-        pico_tree_insert(artree, nsec);
+    	if(pico_tree_insert(artree, nsec) ){
+			if(pico_err != PICO_ERR_ENOMEM){
+				pico_err = PICO_ERR_EINVAL;
+			}
+			return -1;
+		}
```

Comment:
I think `nsec` is allocated in `pico_mdns_gen_nsec_record` (correct me if I'm wrong), so you'll probably need to free it as well.